### PR TITLE
Rename `IS_IN_DATABRICKS_MODEL_SERVING_ENV` to `IS_IN_DB_MODEL_SERVING`

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/12064?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12064/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12064
```

</p>
</details>

### What changes are proposed in this pull request?

The env var `IS_IN_DATABRICKS_MODEL_SERVING_ENV` caused issue in JSL model loading (ES-1080421) due to the incorrect check using env var name (https://github.com/JohnSnowLabs/johnsnowlabs/pull/1222. We've filed a patch for new MLflow version and JSL, but we also need to rename the env var to address the issue for existing models.

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/utils/databricks_utils.py [hunk 1]
```diff
@@ -171,7 +171,17 @@ def is_in_databricks_job():
...
+    Check if the code is running in Databricks Model Serving environment.
+    The environment variable set by Databricks when starting the serving container.
+    """
+    return (
+        os.environ.get("IS_IN_DB_MODEL_SERVING_ENV", "false").lower() == "true"
+        # Checking the old env var name for backward compatibility. The env var was renamed once
+        # to fix a model loading issue, but we still need to support it for a while.
+        # TODO: Remove this once the new env var is fully rolled out.
+        or os.environ.get("IS_IN_DATABRICKS_MODEL_SERVING_ENV", "false").lower() == "true"
+    )
```
Comment (harupy) [inline mlflow/utils/databricks_utils.py:184]: can this be refactored like this?

```suggestion
    val = (
        os.environ.get("IS_IN_DB_MODEL_SERVING_ENV")
        # <comment>
        or os.environ.get("IS_IN_DATABRICKS_MODEL_SERVING_ENV")
        or "false"
    )
    return val.lower() == "true"
```

Comment (B-Step62) [reply mlflow/utils/databricks_utils.py:184]: Yup sounds good!

**Comment by harupy** [comment]
There are a few more environment variables containing `DATABRICKS` 

https://github.com/mlflow/mlflow/blob/12898b30d5a0b55fccd62a2ef9a11b72fa45ec53/tests/utils/test_databricks_utils.py#L131-L135

I'm not sure if they are relevant.

**Review by harupy — APPROVED** [review]
The code LGTM! Have other folks also agreed on the renaming?

**Comment by B-Step62** [comment]
> The code LGTM! Have other folks also agreed on this?

Let's see inference team's opinion in the rag-or-tracing-xxx channel, hopefully there will be a major objection as the renaming idea was originally came from them🙂 Also I've synced with RAG folks separately.

**Comment by harupy** [comment]
sg!

File: mlflow/utils/databricks_utils.py [hunk 2]
```diff
@@ -171,7 +171,19 @@ def is_in_databricks_job():
...
-    return "IS_IN_DATABRICKS_MODEL_SERVING_ENV" in os.environ
+    """
+    Check if the code is running in Databricks Model Serving environment.
+    The environment variable set by Databricks when starting the serving container.
+    """
+    val = (
+        os.environ.get("IS_IN_DB_MODEL_SERVING_ENV")
+        # Checking the old env var name for backward compatibility. The env var was renamed once
+        # to fix a model loading issue, but we still need to support it for a while.
+        # TODO: Remove this once the new env var is fully rolled out.
```
Comment (WeichenXu123) [inline mlflow/utils/databricks_utils.py:182]: do we have internal ticket to track the TODO (as a task reminder )

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Review by BenWilson2 — APPROVED** [review]

File: mlflow/utils/databricks_utils.py (hunk 2 above)
Comment (BenWilson2) [reply mlflow/utils/databricks_utils.py:182]: I filed a cleanup ticket for this and the safe flag special handling for the legacy env var naming

**Review by victorsun123 — APPROVED** [review]

File: mlflow/utils/databricks_utils.py (hunk 2 above)
Comment (B-Step62) [reply mlflow/utils/databricks_utils.py:182]: Created ML-41733 and linked to the original ticket!

**Pull request merged**